### DISCUSSION TITLE
exporting .pt model into int8 quantized .engine model

### DISCUSSION THREAD

**Discussion by ausollet** [ultralytics/yolov5#6493] [Q&A] [answered]
Hey, not sure if this has already been asked, but is it possible to export pyTorch .pt yolov5 models to tensorRT .engine int8 models.

**Comment by glenn-jocher** [comment, marked as answer]
@ausollet see export.py argparser for export arguments:

https://github.com/ultralytics/yolov5/blob/6445a8137e87f67cf3275c70e3585f634260417b/export.py#L479-L508

**Reply by ausollet** [reply]
even after specifying --int8, the engine comes out as such:
TensorRT: Network Description:
TensorRT:       input "images" with shape (1, 3, 640, 640) and dtype DataType.FLOAT
TensorRT:       output "output" with shape (1, 25200, 85) and dtype DataType.FLOAT
TensorRT:       output "467" with shape (1, 3, 80, 80, 85) and dtype DataType.FLOAT
TensorRT:       output "533" with shape (1, 3, 40, 40, 85) and dtype DataType.FLOAT
TensorRT:       output "599" with shape (1, 3, 20, 20, 85) and dtype DataType.FLOAT
TensorRT: building FP32 engine in yolov5/yolov5m.engine

**Reply by glenn-jocher** [reply]
@ausollet argparser help states that --int8 only applies to CoreML/TF
https://github.com/ultralytics/yolov5/blob/6445a8137e87f67cf3275c70e3585f634260417b/export.py#L490

**Reply by ausollet** [reply]
Then how does one go about doing the same for .engine tensorRT? Is the support available only for FP32 and FP16 as of now?

**Reply by ausollet** [reply]
Just changing the config flag to int8 gave rise to errors.
Am I missing something? Do I need to define a calibrator too for int8 conversion?

**Reply by glenn-jocher** [reply]
@ausollet the only available TRT arguments are:

<img width="886" alt="Screen Shot 2022-02-02 at 1 01 19 PM" src="https://user-images.githubusercontent.com/26833433/152149941-05044c3d-bdc6-4b68-b427-ab7581683f84.png">

**Reply by glenn-jocher** [reply]
@ausollet --half is also compatible with TRT export:
https://github.com/ultralytics/yolov5/blob/4c409332667477560200958b513b958bb8fdef71/export.py#L177

**Reply by ausollet** [reply]
Ok, so basically from what I understand, as of now in export.py TRT .engine can only be either FP32 or FP16 (half).
int8 support doesn't exist yet.
Point me out of I'm incorrect.

**Reply by glenn-jocher** [reply]
@ausollet yes that's correct! I'm not a TRT expert so if there's a way to do this please help by submitting a PR.

Please see our ✅ [Contributing Guide](https://github.com/ultralytics/yolov5/blob/master/CONTRIBUTING.md) to get started.
